mesonpy/__init__.py

```diff
@@ -623,6 +623,18 @@ def build_editable(self, directory: Path, verbose: bool = False) -> pathlib.Path
         }
         rebuild_commands = [list(cmd) for cmd in self._project.build_commands(install_path)]
 
+        # XXX: Meson is messing up the install paths on macOS, let's just try
+        #      searching for site-packages directories in the install paths and
+        #      add matches to the import paths
```

Comment:
Ugh? The install paths used by Meson are the ones specified in `sysconfig`. Why do you think Meson is doing something wrong? Please note that the installation directory schema used on macOS is different when installing in an virtual env.

Reply:
> The install paths used by Meson are the ones specified in `sysconfig`

They aren't. They are based on `sysconfig` but they're processed in some cases, which is what seems I am running into here.

https://github.com/mesonbuild/meson/blob/b249470a4c1de39a0a56d9237f767fe04d6efa05/mesonbuild/modules/python.py#L470

I have already recommended that Meson start trying to find Python interpreter in the target prefix, instead of finding *some* interpreter in `PATH` and trying to derive the paths by introspecting that interpreter and trying to override the prefix, Python paths are just too fragile for that to work reliably. Even the `install_env` is utterly conceptually broken, unfortunately layout paths are just not portable between interpreters. After my previous interactions with the maintainers, I am not really motivated to try to fix Python path handling on Meson.

That said, I am not sure if the current issue I am running into is the result of the prefix overriding logic or something else, what I know is that the Meson is trying to install to are definitely not the ones indicated by `sysconfig`.

Debugging shows a path mismatch on both Homebrew Python 3.11, and Apple Python.

Here are the paths on Homebrew Python 3.11:

Meson install path
```
/opt/homebrew/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/opt/homebrew/lib/python3.11/site-packages
```

sysconfig path
```
/opt/homebrew/lib/python3.11/site-packages
```

Reply:
> I have already recommended that Meson start trying to find Python interpreter in the target prefix, instead of finding some interpreter in PATH and trying to derive the paths by introspecting that interpreter and trying to override the prefix, Python paths are just too fragile for that to work reliably.

That does sound sensible to me. It'd be great if there were an upstream issue for this, and the XXX comment referring to it.

Reply:
> Even the install_env option is utterly conceptually broken, unfortunately layout paths are just not portable between interpreters.

... Then it's a good thing thats not really what Meson is trying to do? But I'm pretty sure we discussed this in the past already, nothing new to be said.

> After my previous interactions with the maintainers, I am not really motivated to try to fix Python path handling on Meson.

There's a non-zero chance you're not the only one burned out from that PR. :shrug:

I have some ideas floating around in the back of my head that may be relevant to this whole topic, but for over a year, I've been discovering so many more interesting and enjoyable things to do for Meson than to discuss python layout paths. Most of those things involve not thinking about the python module at all. Maybe next year. Maybe not.

Considering we have to work with the `sysconfig` API of all things :P it may make sense to defer all changes until sysconfig can actually handle the situation. Hopefully sysconfig can at least help resolve the remaining holdouts for porting off of distutils before 3.12 final lands.